source/_posts/2016-10-18-Understanding-code-signing-for-iOS-apps.md

```diff
@@ -0,0 +1,91 @@
+
+
+### App id
+The app id is the unique identifier of your app. It consists of a team id, generated by Apple (you don't have any control over it) and your app's bundle id (`com.youcompany.yourapp`, for example). 
+
+There can also be wildcard app ids: `com.yourcompany.*`. Those will match on multiple bundle ids. 
+
+Generally, you will have an explicit app id, not a wildcard one.
+
+### Certificates
+You have probably already noticed that in order to create a certificate in Apple's developer portal, you need to upload a Certificate Signing Request. You can generate this CSR from the Keychain, and this CSR contains a private key.
+
+Then on the developer portal, you can create a certificate using this CSR. 
+
+There can be multiple types of certificates. The most common are:
+
+* Development (iOS App Development) - You need those to run your app on a device from Xcode. 
+* Distribution (App Store and Ad Hoc) - You need those to be able to distribute your app through the App Store or Ad Hoc
+* APNS (Apple Push Notification Service)  - You need those to be able to send push notifications to your app. Unlike Development or Distribution certificates, APNS certificates are associated with an app id. There can be 2 types of APNS certificates, for development - Apple Push Notification service SSL (Sandbox), and for production - Apple Push Notification service SSL (Sandbox & Production). You need to create both of them if you want push notifications to work on both debug and distribution builds. 
+
+### Devices
+You can add up to 100 devices per product family per membership year to your account. 100 iPhones, 100 iPads, 100 iPod Touch, 100 Apple Watches and 100 Apple TVs. To add a device to your account, you need to add its unique device ID. You can easily find that in Xcode, or (a bit more complicated) in iTunes. A detailed guide on how to add devices to your account can be found [here](https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppDistributionGuide/MaintainingProfiles/MaintainingProfiles.html#//apple_ref/doc/uid/TP40012582-CH30-SW10).
+
+### Provisioning profiles
+The provisioning profile is what associates an app id with a certificate and, for development or ad hoc distribution, with some devices. You create the provisioning profiles on the Apple developer portal and you download them in Xcode.
+
+### Usage
+
+After you have created all these, you can then go to Xcode, add your certificates, refresh your provisioning profiles and then select the provisioning profile you want. You can then select the desired signing identity (based on the certificate associated with it) from that provisioning profile.
+
+### F.A.Q.
+
+Over the years working in iOS development, I've asked and I've been asked many questions about code signing. Here are some of them.
+
+- 
+  **Q**: I have downloaded the provisioning profiles and certificates from the developer portal, but I can't sign the app. 
```

Comment:
I've seen that MacDown renders them on the same line, but they're rendered correctly on the website 
![image](https://cloud.githubusercontent.com/assets/844175/19514410/af97c108-95f3-11e6-8be4-5228900ce419.png)

Anyway, I'll add the 2 spaces so that MacDown doesn't complain :D